### PR TITLE
Add all features unit test job

### PR DESCRIPTION
**Pull request by saschagrunert**
Follow-up of https://github.com/containers/oci-spec-rs/pull/15#discussion_r691286220

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -95,3 +95,20 @@ jobs:
...
+        run: cargo generate-lockfile
+      - name: Setup Cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-test-${{ hashFiles('**/Cargo.lock') }}
+      - run: cargo test --all-features
```
Comment (utam0k) [inline .github/workflows/ci.yml:114]: Wouldn't it be more appropriate to measure the code coverage in this job?

Comment (saschagrunert) [reply .github/workflows/ci.yml]: I think so, switching the flags as suggested :+1:

**Comment by saschagrunert** [comment]
Running both tests (with and without all features) in sequence works too :)

**Review by utam0k — APPROVED** [review]
lgtm

**Pull request merged**